src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,210 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.DateTimeException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Handles parsing of user input and execution of commands in the Meow application.
+ * Supports various commands such as adding/deleting tasks, marking/unmarking tasks,
+ * as well as viewing outstanding tasks.
+ */
+public class Parser {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```

Comment:
Good to see making a frequently used variable constant and it is following its naming convention.